### PR TITLE
Try using Jenkins Pipelines with Jenkinsfile.

### PR DESCRIPTION
**Pull request by seantalts**
This should trigger this job:

http://d1m1s1b1.stat.columbia.edu:8080/job/Ignore%20-%20Sean%20-%20MathMulti/

I think we can replace our existing Math and Math Pull Request jobs with this Jenkinsfile and the build above (renamed).

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by seantalts** [comment]
(currently failing because develop is broken from here: http://d1m1s1b1.stat.columbia.edu:8080/job/Math%20-%20Tests%20-%20Header/249/

File: make/os_mac [hunk 1]
```diff
@@ -5,7 +5,6 @@
 #   - GTEST_CXXFLAGS
 
 ifeq (g++,$(CC_TYPE))
-  CXXFLAGS += -Wno-unused-function
```
Comment (syclik) [inline make/os_mac]: did you intend to remove this line?

Comment (seantalts) [reply make/os_mac]: Woops, yes but maybe it should have been a separate PR - I meant to take it out in the last PR where I moved this warning and another into the default options

**Comment by syclik** [comment]
@seantalts, what's the state of this PR?

**Comment by seantalts** [comment]
I still want to get through an entire run - the last failure was because the mac pro ran out of disk space, so I set the old distribution tests job to delete their workspace when they're done and we can re-run this.

**Comment by syclik** [comment]
@seantalts, ping me if you need anything on this PR.

**Comment by seantalts** [comment]
Hey @syclik, to get this working I had to pipe the output of the distribution tests to a file on disk. I can have the contents `tail`'d if the tests fail. For some reason some part of the pipelines can't handle the 2GB of console output from the distribution tests :'( 

I'm going to go ahead and merge this and disable (but not delete) the other jobs from going automatically once this builds in develop.

**Comment by syclik** [comment]
Cool. I have no idea why there’d be a GB limitation, but ok.

**Comment by seantalts** [comment]
I’m guessing it’s a bug due to a timeout hidden somewhere... something like
that.

**Pull request merged**